src/VisualStudio/Core/Def/Implementation/Log/VisualStudioErrorLogger.cs

```diff
@@ -4,23 +4,85 @@
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CodeRefactorings;
 using Microsoft.CodeAnalysis.ErrorLogger;
 using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.VisualStudio.LanguageServices.Implementation.Watson;
 using Microsoft.VisualStudio.Shell;
 
 namespace Microsoft.VisualStudio.LanguageServices.Implementation.Log
 {
-    [ExportWorkspaceService(typeof(IErrorLoggerService), ServiceLayer.Host),  Export(typeof(IErrorLoggerService)), Shared]
+    [ExportWorkspaceService(typeof(IErrorLoggerService), ServiceLayer.Host), Export(typeof(IErrorLoggerService)), Shared]
     internal class VisualStudioErrorLogger : IErrorLoggerService
     {
-        public void LogError(string source, string message)
+        public void LogException(object source, Exception exception)
         {
-            ActivityLog.LogError(source, message);
+            var name = source.GetType().Name;
+            ActivityLog.LogError(name, ToLogFormat(exception));
+
+            if (ShouldReportCrashDumps(source))
+            {
+                using (var report = WatsonErrorReport.CreateNonFatalReport(new ExceptionInfo(exception, name)))
```

Comment:
any reason not using FatalError static type we use to abstract out actual watson?